core/src/main/java/org/jdbi/v3/core/statement/SqlStatements.java

```diff
@@ -32,24 +32,19 @@
  */
 public final class SqlStatements implements JdbiConfig<SqlStatements> {
 
-    private final Map<String, Object> attributes;
-    private TemplateEngine templateEngine;
-    private SqlParser sqlParser;
-    private SqlLogger sqlLogger;
-    private Integer queryTimeout;
-    private boolean allowUnusedBindings;
+    // cannot be a ConcurrentHashMap because of null keys
+    private final Map<String, Object> attributes = Collections.synchronizedMap(new HashMap<>());
```

Comment:
Exhibit A for thread safety of config being a poor API choice. Make everything slower "just in case."